antelope_contracts/contracts/erc20/src/erc20.cpp

```diff
@@ -86,6 +86,22 @@ void erc20::upgrade() {
     });
 }
 
+void erc20::upgradeto(std::string impl_address) {
+    require_auth(get_self());
+    auto address_bytes = from_hex(impl_address);
+    eosio::check(!!address_bytes, "implementation address must be valid 0x EVM address");
+    eosio::check(address_bytes->size() == kAddressLength, "invalid length of implementation address");
+
+    uint64_t id = 0;
+    impl_contract_table_t contract_table(_self, _self.value);
+
+    contract_table.emplace(_self, [&](auto &v) {
```

Comment:
It would be better to provide a more meaningful error message if someone calls `upgradeto` after the table entry with `id` of 0 was already created (e.g. due to a prior call of `upgradeto` or `upgrade`. Right now it goes deep into chainbase code before it would give an confusing error about a uniqueness constraint being violated.

I won't block this PR from being merged in because of this. But maybe @yarkinwho can add that check to this action with a better error message when he adds the tests in his PR (and includes in those tests checks that calling `upgrade` or `upgradeto` after they have already been called will fail as expected).